File: app/src/main/java/mk/android/com/canvasdrawview/adapter/RecyclerViewEmptyObserver.java
package mk.android.com.canvasdrawview.adapter;

import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.view.View;
import android.widget.TextView;

/**
 * Created by Mayuri Khinvasara on 03,December,2018
 */
public class RecyclerViewEmptyObserver extends RecyclerView.AdapterDataObserver {

    private static final String LOG_TAG = RecyclerViewEmptyObserver.class.getSimpleName();
    private TextView emptyView;
    private RecyclerView recyclerView;


    /**
     * Constructor to set an Empty View for the RV
     */
    public RecyclerViewEmptyObserver(RecyclerView rv, TextView ev) {
        this.recyclerView = rv;
        this.emptyView    = ev;
        checkIfEmpty();
    }

    /**
     * Check if Layout is empty and show the appropriate view
     */
    private void checkIfEmpty() {

        if (emptyView != null && recyclerView.getAdapter() != null) {

            boolean emptyViewVisible = recyclerView.getAdapter().getItemCount() == 0;
            Log.d(LOG_TAG, " Enabling empty view for list : No data found " );
            emptyView.setVisibility(emptyViewVisible ? View.VISIBLE : View.GONE);
            recyclerView.setVisibility(emptyViewVisible ? View.GONE : View.VISIBLE);
        }
    }


    /**
     Abstract method implementations
     */
    @Override
    public void onChanged() {
        checkIfEmpty();
    }

    @Override
    public void onItemRangeInserted(int positionStart, int itemCount) {
        checkIfEmpty();
    }

    @Override
    public void onItemRangeRemoved(int positionStart, int itemCount) {
        checkIfEmpty();
    }
}

File: app/src/main/java/mk/android/com/canvasdrawview/adapter/RecyclerViewEmptyObserver.kt
package mk.android.com.canvasdrawview.adapter

import android.support.v7.widget.RecyclerView
import android.util.Log
import android.view.View
import android.widget.TextView

/**
 * Created by Mayuri Khinvasara on 03,December,2018
 */
class RecyclerViewEmptyObserver
/**
 * Constructor to set an Empty View for the RV
 */
(private val recyclerView: RecyclerView, private val emptyView: TextView?) : RecyclerView.AdapterDataObserver() {


    init {
        checkIfEmpty()
    }

    /**
     * Check if Layout is empty and show the appropriate view
     */
    private fun checkIfEmpty() {

        if (emptyView != null && recyclerView.adapter != null) {

            val emptyViewVisible = recyclerView.adapter!!.itemCount == 0
            Log.d(LOG_TAG, " Enabling empty view for list : No data found ")
            emptyView.visibility = if (emptyViewVisible) View.VISIBLE else View.GONE
            recyclerView.visibility = if (emptyViewVisible) View.GONE else View.VISIBLE
        }
    }


    /**
     * Abstract method implementations
     */
    override fun onChanged() {
        checkIfEmpty()
    }

    override fun onItemRangeInserted(positionStart: Int, itemCount: Int) {
        checkIfEmpty()
    }

    override fun onItemRangeRemoved(positionStart: Int, itemCount: Int) {
        checkIfEmpty()
    }

    companion object {

        private val LOG_TAG = RecyclerViewEmptyObserver::class.java.simpleName
    }
}
